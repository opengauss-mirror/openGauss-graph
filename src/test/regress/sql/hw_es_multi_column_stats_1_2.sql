--==========================================================
--==========================================================
\set ECHO all
set current_schema = hw_es_multi_column_stats;
set default_statistics_target=-2;

--========================================================== alter table
analyze t1_2;
analyze t1r_2;

alter table t1_2 add statistics ((a, b));
alter table t1_2 add statistics ((b, c));
alter table t1_2 add statistics ((a, d));
alter table t1_2 add statistics ((b, d));
alter table t1_2 add statistics ((c, d));
alter table t1_2 add statistics ((b, c, d));

alter table t1r_2 add statistics ((a, b));
alter table t1r_2 add statistics ((b, c));
alter table t1r_2 add statistics ((a, d));
alter table t1r_2 add statistics ((b, d));
alter table t1r_2 add statistics ((c, d));
alter table t1r_2 add statistics ((b, c, d));

select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1_2' order by attname;
select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1r_2' order by attname;

analyze t1_2;
analyze t1r_2;

select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1_2' order by attname;
select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1r_2' order by attname;
