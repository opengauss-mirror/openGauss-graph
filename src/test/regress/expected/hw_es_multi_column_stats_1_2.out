--==========================================================
--==========================================================
\set ECHO all
set current_schema = hw_es_multi_column_stats;
set default_statistics_target=-2;
--========================================================== alter table
analyze t1_2;
analyze t1r_2;
alter table t1_2 add statistics ((a, b));
alter table t1_2 add statistics ((b, c));
alter table t1_2 add statistics ((a, d));
alter table t1_2 add statistics ((b, d));
alter table t1_2 add statistics ((c, d));
alter table t1_2 add statistics ((b, c, d));
alter table t1r_2 add statistics ((a, b));
alter table t1r_2 add statistics ((b, c));
alter table t1r_2 add statistics ((a, d));
alter table t1r_2 add statistics ((b, d));
alter table t1r_2 add statistics ((c, d));
alter table t1r_2 add statistics ((b, c, d));
select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1_2' order by attname;
        schemaname        | tablename | attname | inherited | null_frac | avg_width | n_distinct | n_dndistinct | most_common_vals | most_common_freqs | most_common_vals_null | most_common_freqs_null | histogram_bounds 
--------------------------+-----------+---------+-----------+-----------+-----------+------------+--------------+------------------+-------------------+-----------------------+------------------------+------------------
 hw_es_multi_column_stats | t1_2      | 1 2     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 1 4     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 2 3     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 2 3 4   | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 2 4     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 3 4     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
(6 rows)

select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1r_2' order by attname;
        schemaname        | tablename | attname | inherited | null_frac | avg_width | n_distinct | n_dndistinct | most_common_vals | most_common_freqs | most_common_vals_null | most_common_freqs_null | histogram_bounds 
--------------------------+-----------+---------+-----------+-----------+-----------+------------+--------------+------------------+-------------------+-----------------------+------------------------+------------------
 hw_es_multi_column_stats | t1r_2     | 1 2     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 1 4     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 2 3     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 2 3 4   | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 2 4     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 3 4     | f         |         0 |         0 |          0 |            0 |                  |                   |                       |                        | 
(6 rows)

analyze t1_2;
analyze t1r_2;
select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1_2' order by attname;
        schemaname        | tablename | attname | inherited | null_frac | avg_width | n_distinct | n_dndistinct |           most_common_vals            |         most_common_freqs         | most_common_vals_null | most_common_freqs_null | histogram_bounds 
--------------------------+-----------+---------+-----------+-----------+-----------+------------+--------------+---------------------------------------+-----------------------------------+-----------------------+------------------------+------------------
 hw_es_multi_column_stats | t1_2      | 1 2     | f         |         0 |         8 |         -1 |            0 |                                       |                                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 1 4     | f         |         0 |         8 |         -1 |            0 |                                       |                                   |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 2 3     | f         |         0 |         8 |   -.555556 |            0 | {"{1,2,3,4}","{1,1,2,2}"}             | {.222222,.222222,.222222,.222222} |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 2 3 4   | f         |         0 |        12 |   -.555556 |            0 | {"{1,2,3,4}","{1,1,2,2}","{1,1,1,1}"} | {.222222,.222222,.222222,.222222} |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 2 4     | f         |         0 |         8 |   -.555556 |            0 | {"{1,2,3,4}","{1,1,1,1}"}             | {.222222,.222222,.222222,.222222} |                       |                        | 
 hw_es_multi_column_stats | t1_2      | 3 4     | f         |         0 |         8 |   -.333333 |            0 | {"{1,2}","{1,1}"}                     | {.444444,.444444}                 |                       |                        | 
(6 rows)

select * from pg_ext_stats where schemaname='hw_es_multi_column_stats' and tablename='t1r_2' order by attname;
        schemaname        | tablename | attname | inherited | null_frac | avg_width | n_distinct | n_dndistinct |           most_common_vals            |         most_common_freqs         | most_common_vals_null | most_common_freqs_null | histogram_bounds 
--------------------------+-----------+---------+-----------+-----------+-----------+------------+--------------+---------------------------------------+-----------------------------------+-----------------------+------------------------+------------------
 hw_es_multi_column_stats | t1r_2     | 1 2     | f         |         0 |         8 |         -1 |            0 |                                       |                                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 1 4     | f         |         0 |         8 |         -1 |            0 |                                       |                                   |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 2 3     | f         |         0 |         8 |   -.555556 |            0 | {"{1,2,3,4}","{1,1,2,2}"}             | {.222222,.222222,.222222,.222222} |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 2 3 4   | f         |         0 |        12 |   -.555556 |            0 | {"{1,2,3,4}","{1,1,2,2}","{1,1,1,1}"} | {.222222,.222222,.222222,.222222} |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 2 4     | f         |         0 |         8 |   -.555556 |            0 | {"{1,2,3,4}","{1,1,1,1}"}             | {.222222,.222222,.222222,.222222} |                       |                        | 
 hw_es_multi_column_stats | t1r_2     | 3 4     | f         |         0 |         8 |   -.333333 |            0 | {"{1,2}","{1,1}"}                     | {.444444,.444444}                 |                       |                        | 
(6 rows)

